test/00/deleteKey.sh

```diff
@@ -27,38 +27,83 @@ pass()
 }
 
 trap "fail" 1 2 3 15
-cat >input.tcl <<EOF
-source $here/test/assert.tcl
-minsky.load $here/examples/GoodwinLinear02.mky
-minsky.canvas.recentre
-proc afterMinskyStarted {} {
-  puts "minsky started"
-  set numItems [model.numItems]
-  set numWires [model.numWires]
-  focus .wiring.canvas
+cat >input.py <<EOF
+import sys
+sys.path.insert(0, '$here')
+from pyminsky import minsky
 
-#  set w [minsky.model.wires.@elem 1]
-#  set x {[\$w.coords.@elem 0]}
-#  set y {[\$w.coords.@elem 1]}
-#  deleteKey \$x \$y
-  deleteKey 415 290
-  assert "[model.numWires]==[expr \$numWires-1]" {test wire deletion}
-  findVariable emprate
-  deleteKey [minsky.canvas.item.x] [minsky.canvas.item.y]
-  assert "[model.numItems]==[expr \$numItems-1]" {test emprate var deletion}
-  # should also be two more wires deleted
-  assert "[model.numWires]==[expr \$numWires-3]" {test attached wires deletion}
+# Load the model and centre the canvas
+minsky.load('$here/examples/GoodwinLinear02.mky')
+minsky.canvas.recentre()
 
-  findVariable Investment
-  canvas.selectAllVariables
-  deleteKey 0 0
-  assert "[model.numItems]==[expr \$numItems-2]" {test selection deletion}
+# Store initial counts of items and wires
+numItems = minsky.model.numItems()
+numWires = minsky.model.numWires()
 
-  tcl_exit
-}
+# Function to find a variable by name
+def findVariable(name):
+    last_error = None
+    for i in range(len(minsky.model.items)):
+        try:
+            if minsky.model.items[i].name() == name:
+                x = minsky.model.items[i].x()
+                y = minsky.model.items[i].y()
+                print(f"Found variable '{name}' at coordinates ({x}, {y})")
+                # Get the item at coordinates
+                if minsky.canvas.getItemAt(x, y):
+                    return (x, y)
+                else:
+                    print(f"Failed to focus item at ({x}, {y}). Trying manual fallback.")
+                    return (x, y)  # Return coordinates anyway for manual handling
+        except Exception as e:
+            last_error = e
+            pass
+    if last_error:
+        print(f"Error while finding variable '{name}' at last known coordinates: {last_error}")
+    return None
+
+# Function to delete an item or wire at a given position
+def deleteKey(x, y):
+    # Check for a wire or item at the coordinates and delete it
+    if minsky.canvas.getWireAt(x, y):
+        minsky.canvas.deleteWire()
+        print(f"Wire deleted at ({x}, {y})")
+    elif minsky.canvas.getItemAt(x, y):
+        item = minsky.canvas.itemFocus()  # Try to focus on the item
```

Comment:
This part is a bit strange - getItemAt populates canvas.item, not canvas.itemFocus. But this doesn't invalidate the test.
`bool getItemAt(float x, float y) override {return (item=itemAt(x,y)).get();}`